src/data.table.h

```diff
@@ -46,6 +46,8 @@ typedef R_xlen_t RLEN;
 // for use with bit64::integer64
 #define NA_INTEGER64  INT64_MIN
 #define MAX_INTEGER64 INT64_MAX
+#define INTEGER64_ASCHAR_LEN 22
+#define INTEGER64_ASCHAR_FMT "%"PRId64""
```

Comment:
This gives a compiler warning on my linux box, but not my windows machine:

```
In file included from utils.c:1:0:
utils.c: In function ‘asCharacterInteger64’:
data.table.h:50:30: warning: format ‘%ld’ expects argument of type ‘long int’, but argument 4 has type ‘long long int’ [-Wformat=]
 #define INTEGER64_ASCHAR_FMT "%"PRId64""
                              ^
utils.c:377:44: note: in expansion of macro ‘INTEGER64_ASCHAR_FMT’
       snprintf(buff, INTEGER64_ASCHAR_LEN, INTEGER64_ASCHAR_FMT, px[i]);
                                            ^~~~~~~~~~~~~~~~~~~~
In file included from dt_stdio.h:31:0,
                 from data.table.h:1,
                 from utils.c:1:
/usr/include/inttypes.h:57:34: note: format string is defined here
 # define PRId64  __PRI64_PREFIX "d"
```

Reply:
Expected format type in bit64 package is "%lld"

Reply:
IIRC we hit a portability issue on CRAN with %lld and switched fairly recently to the %PRId64 approach. might be something we suggest to bit64 to fix